Doc/whatsnew/3.8.rst

```diff
@@ -697,6 +714,14 @@ Changes in the Python API
   environment variable and does not use :envvar:`HOME`, which is not normally
   set for regular user accounts.
 
+* DLL dependencies for extension modules and DLLs loaded with :mod:`ctypes` on
+  Windows are now resolved more securely. Only the system paths, the directory
+  containing the DLL or PYD file, and directories added with
+  :func:`~os.add_dll_directory` are searched for load-time dependencies.
+  Specifically, :envvar:`PATH` is no longer used, and modifications to this
+  environment variable will no longer have any effect on normal DLL resolution.
+  (See :issue:`36085`.)
+
```

Comment:
Can most of this explanation be repeated in the documentation for `add_dll_directory`? It will greatly aid transition to the new paradigm. Users are more likely to look in the `os` documentation for this information than in the release notes.

Reply:
Sure, though it'll come with some kind of version specific tag to make clear that it's best practice rather than a Python guarantee.

Reply:
Something like "After this call and perhaps before it, :envvar:`PATH` is not used, and modifications to this environment variable will not have any effect on DLL resolution. See <> for more information." would be enough.

Reply:
The call is irrelevant - we just never use PATH or cwd anymore, and the issue is linked for more information. The updated text should make this a little clearer.